Cue/app/actions/library.js

```diff
@@ -17,4 +19,80 @@ async function loadLibrary(): PromiseAction {
   };
 }
 
-module.exports = { loadLibrary };
+function deleteDeck(uuid: string) : Action {
+  return {
+    type: 'DECK_DELETED',
+    uuid
+  }
+}
+
+function createDeck(name: string) : Action {
+  let deck = {
+    name,
+    uuid: uuidV4(),
+    public: false,
+    cards: [],
+    owner: CueApi.userId,
+    accession: 'private',
+  }
+  return {
+    type: 'DECK_CREATED',
+    deck,
+  }
+}
+
+function editDeck(change) : Action {
+  return {
+    type: 'DECK_EDITED',
+    change
+  }
+}
+
+//attempts to sync each deck once, returns list of failed syncs
+function syncLibrary(localChanges): ThunkAction {
+  let failedSyncs = []
+  return (dispatch, getState) => {
+    let promises = []
+    localChanges.forEach(change =>{
+      promises.push(dispatch(syncDeck(change))
+        .catch(e => {
+          console.info('sync failed : ', change)
+          failedSyncs.push(change)
+        }))
+    })
+    Promise.all(promises).then(() => {
+      if (failedSyncs.length == 0) dispatch(loadLibrary())
+    })
+    console.log('failed syncs:  ', failedSyncs)
+    return failedSyncs
+  };
+}
+
+// deck should only contain changes in the deck to avoid sending excess data
+// ex: { uuid: string, action 'add', name: string } if a deck was added
+// ex: { uuid: string, action 'edit', cards: [{action: 'edit', card-id: string, front: "fox"}] } if 1 card in deck was changed
```

Comment:
I think its actually `card_id` and also the card would have to include the `back` as well, so:
```
{ action: 'edit', card_id: string, front: 'fox', back: '20XX' }
```